tests/Functional/SelfUpdateTest.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Manalize project.
+ *
+ * (c) Manala <contact@manala.io>
+ *
+ * For the full copyright and license information, please refer to the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Manala\Manalize\Tests\Functional;
+
+use Manala\Manalize\Command\Setup;
+use Manala\Manalize\Application;
+use Symfony\Component\Console\Tester\CommandTester;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Process\Process;
+
+class SelfUpdateTest extends \PHPUnit_Framework_TestCase
+{
+    private static $cwd;
+
+    public static function setUpBeforeClass()
+    {
+        $cwd = sys_get_temp_dir().'/Manala';
```

Comment:
```diff
- $cwd = sys_get_temp_dir().'/Manala';
+ $cwd = sys_get_temp_dir().'/Manala/tests/selfupdate';
```

to avoid any issue with the tmp dir already used by another process, as I may have encountered with the `DiffTest`.